site/content/blog/2021-11-26-launching-browsers-with-extensions/index.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Launching Browsers With Extensions"
+summary: >
+  You can now launch your favourite browsers from ZAP with your favourite extensions.
+type: post
+tags:
+- blog
+- browser
+date: "2021-11-26"
+authors:
+    - simon
+---
+Configuring your browser to proxy through ZAP has been an unnecessary chore since 2017 when we introduced
+[Browser Launch](/blog/2017-08-22-zap-browser-launch/).
+
+For those of you who are not aware of this feature it allows you to launch modern browsers (i.e. Firefox and Chrome) from ZAP -
+these are preconfigured to proxy via ZAP and ignore certificate exceptions. So no extra setting up is needed at all.
+
+OK, thats not strictly true - these browsers are launched with new profiles which meant that if you wanted to use any browser extensions
+then you would need to install them every time you launch a new browser. Or go back to using your existing browsers and having to set them up to proxy through ZAP again.
+
+But that is no longer the case - you can now tell ZAP which browser extensions you want to use via the Selenium Options screen:
+
+![Selenium Options](images/selenium-options.png)
+
+<TODO add image>
+
+Clicking on the "Add" button will bring up a dialog which will allow you to select the extension file you want to add.
+The file must be accessible locally and have a ".xpi" extension for Firefox or a ".crx" extension for Chrome.
+
+Browser add-ons can be enabled and disabled via this panel, only enabled extensions will be installed when you launch browsers from ZAP.
+
+### Firefox Extensions
+
+Firefox extensions can be found on https://addons.mozilla.org/en-GB/firefox/ - just search for the extension you want, right click on the blue
```

Comment:
Does this need to be `<https://addons.mozilla.org/>` so that it actually gets linkified?

Also seems the locale and firefox can be dropped.

Reply:
It will be a link automatically.